### PR TITLE
Add initial setup for tests with `vitest`

### PR DESCRIPTION
**Pull request by zaiste**
Related to #234

### CODE REVIEW

**Comment by lkostrowski** [comment]
I have peer deps problem on this branch

```
 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

.
├─┬ eslint-config-airbnb
│ └── ✕ missing peer eslint-plugin-react-hooks@^4.3.0
└─┬ aws-amplify
  └─┬ @aws-amplify/analytics
    └─┬ @aws-amplify/cache
      └─┬ @aws-amplify/core
        └─┬ @aws-sdk/client-cloudwatch-logs
          └─┬ @aws-sdk/middleware-retry
            └─┬ react-native-get-random-values
              └── ✕ missing peer react-native@>=0.56
Peer dependencies that should be installed:
  eslint-plugin-react-hooks@^4.3.0  react-native@>=0.56       
```

File: test/vercel.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+const vercel = new Vercel('asdf');
+
+beforeAll(() => server.listen({ onUnhandledRequest: 'error' }));
+afterAll(() => server.close());
+afterEach(() => server.resetHandlers());
+
+describe('Vercel API', () => {
+  it('returns `id` for single project', async () => {
+    // FIX not intercepted
+    const r = await fetch('https://rest.example/path/to/posts');
```
Comment (lkostrowski) [inline test/vercel.test.ts:46]: fetch is not polyfilled and test doesnt pass

File: test/vercel.test.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { setupServer } from 'msw/node';
+import {
+  afterAll,
+  afterEach,
+  assert,
+  beforeAll,
+  describe,
+  expect,
+  it,
+} from 'vitest';
```
Comment (lkostrowski) [inline test/vercel.test.ts:11]: this is great, far better than jest globals

**Comment by lkostrowski** [comment]
Currently tests are in `test/` dir. What do you think about having them near files? IMO its more natural, that each file has it's own test, as a cohesive unit. 

`/test/` or `/e2e` are app-level tests, local - units

File: test/vercel.test.ts (hunk 1 above)
Comment (lkostrowski) [reply test/vercel.test.ts:46]: add `import fetch from 'node-fetch';`

**Comment by lkostrowski** [comment]
I tested it and read a little about vitest. License looks good, works out of the box. We don't have any tests now to check more serious scenarios, but I believe it can be configured easily. There is easy migration path to jest if we have to.

So 🟢 to finish this PR and let's write some tests 💪 

One comment on testing itself - CLI is pretty big. I went over the codebase. It's not easily testable now. I think we should start with integrations like your example - test some scenarios with mocked responses. In the meantime, we can think about e2e.

But, we should also introduce "normal" unit tests. This will require some cleanup in code. Maybe if we add new functionalities, we can create them in separate files (I'd drop huge "util.ts" files etc) and each file can have it's dedicated test suite. 

Of course - incremental, do not refactor entire app or write tests only for a month. Writing tests for found bugs and for new features can be a good start

**Review by lkostrowski — APPROVED** [review]

**Review by 2can — APPROVED** [review]

**Pull request merged**